### ISSUE TITLE
Speak at Volume no longer sets volume

### ISSUE THREAD

**Issue by alexruffell** [bbindreiter/node-red-contrib-alexa-remote2-applestrudel#22]
I believe I lost ability to play an alert at a specific volume right after upgrading to 5.0.7. If I just use the node to change volume it works fine, but if I try to set volume and play the message, the message is played at whatever volume the echo was set at. Also, the beep that plays before and after the message is no longer sounding. 

![image](https://user-images.githubusercontent.com/4006944/139540325-7dd3c219-522f-443e-a6e3-aed509929c0f.png)

**Comment by bbindreiter** [comment]
Hi! To rule out any problems with the payload (shouldn't it be `msg.payload.volume`), can you please try to set it to some number and report back whether it works?

![image](https://user-images.githubusercontent.com/3252801/142744137-465d9ef6-028b-4c1c-b348-a89405749e8f.png)

**Comment by alexruffell** [comment]
@bbindreiter - The issue persisted for some amount of time and then went away on it own. I figured it was something on Alexa's end. 

As for the msg.volume I use, I set it to a value in a change node as shown below:

<img width="218" alt="image" src="https://user-images.githubusercontent.com/4006944/142747067-347f4363-0866-4910-b43b-ae6d34ce9b1d.png">

<img width="281" alt="image" src="https://user-images.githubusercontent.com/4006944/142747072-56b3e255-72dd-41c7-844c-49ec33f04a30.png">

On another note, one of the Echo's I use for the announcements has a ? as it's icon (see below). Is this an indication of a problem? If so should I report it on another ticket?

<img width="437" alt="image" src="https://user-images.githubusercontent.com/4006944/142747107-0482b844-4a16-4148-8d5f-e4e7b858bad9.png">

**Comment by bbindreiter** [comment]
Thanks for reporting back! Regarding the icon, the deviceFamily is just not defined (yet), hence the fallback question mark. What Echo device is this exactly?

**Comment by alexruffell** [comment]
@bbindreiter Kitchen, Office and Study are Echo Show 5 (1st Gen, 2019 release). Ha Tablet Echo Show is a Lenovo Tablet that has Alexa mode making it act as an Echo Show. The others are apps.

**Comment by nberenbold** [comment]
Having the same issue so far on a Echo Dot 4. Gen on a fresh installation

**Comment by bbindreiter** [comment]
Guys, you know what. I just realized that I have been having the same issue on some of my Alexa commands. Will see if I can find out more tomorrow.

**Comment by bbindreiter** [comment]
I can reproduce it with Speak at Volume Announcement but seems to work for Regular. As a workaround you can put a Volume Set Node before it.

**Comment by alexruffell** [comment]
Mine seems to be working. It had stopped for a few days when I reported it but have not noticed it since. The only issue I've had since reporting has been a couple instances of me having to edit the Alexa Account to force an "init by proxy" otherwise noting plays on my Echos... but that is likely unrelated to this issue/thread.

**Comment by nberenbold** [comment]
> I can reproduce it with Speak at Volume Announcement but seems to work for Regular. As a workaround you can put a Volume Set Node before it.

I got this issue sometimes on regular but all the time on announce.
Working with a volume node before is kinda tricky since he takes too long sometimes if i would like to reset it afterwards to what it was before (getting errors if i try too fast or something like that)

**Comment by alexruffell** [comment]
> (getting errors if i try too fast or something like that)

Are you referring to the "No JSON..." errors? I get those often too. When you trigger it again it almost always works without the same error. I too figured it was related to speed and added a few delays to see if it helps.

**Comment by bbindreiter** [comment]
From what I can see in the code `Speak At Volume` is doing nothing else but fetching the volume for all devices, followed by a regular `Speak`, and then restores the previously set volume on all devices.

**Comment by alexruffell** [comment]
@bbindreiter Strange, I thought it was working properly at some point whereas it was saving the current volume, setting the new volume for the message, and restoring the previous current volume it saved. Is it not doing so any longer? If not, what is the purpose of what it is doing now?

**Comment by bbindreiter** [comment]
Is it working for you guys again? For me it magically resolved itself.

**Comment by alexruffell** [comment]
The volume issue, as it was when I opened this ticket, seems to have resolved itself. However, my automation setting the volume is failing, along with automations that play messages because my initialization is failing frequently. I used to use the settings in the account that store the cookie in a file, but then moved to chron triggering the initialize node every morning at 6am hoping it would make things better. I still have days where nothing works until I manually trigger the init. Maybe this should be in a different ticket... Thoughts?
